AnnService/src/IndexBuilder/main.cpp

```diff
@@ -87,9 +87,21 @@ int main(int argc, char* argv[])
             fprintf(stderr, "Failed to read input file.\n");
             exit(1);
         }
-        code = indexBuilder->BuildIndex(vectorReader->GetVectorSet(), vectorReader->GetMetadataSet());
-        indexBuilder->SaveIndex(options->m_outputFolder);
+
+        p_vectorSet = vectorReader->GetVectorSet();
```

Comment:
Is there an issue with using the out-of-scope `auto vectorReader` here? I wasn't sure about the lifetime of it, whether there could be an issue. Again, new to all this.